_data-prepper/pipelines/configuration/processors/mutate-event.md

```diff
@@ -0,0 +1,312 @@
+
+```json
+{"message": "value"}
+```
+
+When you run this processor, it parses the message into the following output:
+
+```json
+{"message": "value", "newMessage": "value"}
+```
+
+> If `newMessage` had already existed, its existing value would have been overwritten with `value`
+
+
+## DeleteEntry
+
+The `DeleteEntry` processor deletes entries in an event.
+
+### Configuration
+
+You can configure the `DeleteEntry` processor with the following options:
+
+Option | Required | Description
+:--- | :--- | :---
+| `with_keys` | Yes | An array of keys of the entries to be deleted. | 
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - delete_entries:
+        with_keys: ["message"]
+  sink:
+    - stdout:
+```
+
+Create the following file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"message": "value", "message2": "value2"}
+```
+
+When you run the `DeleteEntry` processor, it parses the message into the following output:
+
+```json
+{"message2": "value2"}
+```
+
+> If `message` does not exist in the event, then no action occurs.
+
+
+## RenameKey
+
+The `rename key` processor renames keys in an event.
+
+### Configuration
+
+You can configure the `rename key` processor with the following options:
+
+Option | Required | Description
+:--- | :--- | :---
+| `entries` | Yes | A list of entries to rename in an event. | 
+| `from_key` | Yes | The key of the entry to be renamed. |
+| `to_key` | Yes | The new key of the entry.
+| `overwrite_if_to_key_exists` | No | When set to a value of`true`, if `to_key` already exists in the event, then the existing value will be overwritten. The default value is `false`.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_keys:
+        entries:
+        - from_key: "message"
+          to_key: "newMessage"
+          overwrite_if_to_key_exists: true
+  sink:
+    - stdout:
+```
+
+Create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"message": "value"}
+```
+
+When run, the processor parses the message into the following output:
+
+```json
+{"newMessage": "value"}
+```
+
+> If `newMessage` already exists, its existing value is overwritten with `value`.
+
+### Special considerations
+
+The renaming operation occurs in a defined order. This means that chaining is implicit with the `RenameKey` processor. See the following `piplines.yaml` file example:
```

Comment:
@oeyh What is the defined order, and why is it important to be defined in this way? Also, what is chaining, and why is it implicit?

Reply:
"defined order" refers to the order of the "from_key" and "to_key" pairs as appeared in the yaml file from top down. 

Look at the yaml file below, there's two "from_key" and "to_key" pairs. They take effect in the order of definition from top to bottom one by one, that is, key will be changed from "message" to "message2" and then from "message2" to "message3". If there're more "from_key" and "to_key" pairs, we will keep going.